src/macaron/dependency_analyzer/cyclonedx.py

```diff
@@ -70,7 +70,7 @@ def get_root_component(root_bom_path: Path) -> Optional[dict | None]:
     try:
         root_bom = deserialize_bom_json(root_bom_path)
         return root_bom.get("metadata").get("component")  # type: ignore
-    except (CycloneDXParserError, KeyError) as error:
+    except (CycloneDXParserError, AttributeError) as error:
```

Comment:
I wonder if we should isolate the try-catch block for AttributeError on the line `return root_bom.get("metadata").get("component")` only? To me, `AttributeError` seems like a very popular Error that could happen because of other reasons too so we should only limit it on the line we are expecting it to happen. This could prevent us from accidentally swallow the `AttributeError` not raised by the usages of chainning `get()` on dictionary.
This comment applies to the other 2 changes as well.

Reply:
This is on purpose, because the SBOMs  can be missing various elements and the blocks are quite isolated for  `AttributeError`s.

Reply:
See commit [877d877](https://github.com/oracle/macaron/pull/329/commits/877d87732d1c47c8fa248b2b066d76aae5712160).